WalletWasabi.Tests/UnitTests/EventSourcing/InMemoryEventRepositoryTests.cs

```diff
@@ -0,0 +1,438 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WalletWasabi.EventSourcing;
+using WalletWasabi.Exceptions;
+using WalletWasabi.Interfaces.EventSourcing;
+using WalletWasabi.Tests.UnitTests.EventSourcing.TestDomain;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace WalletWasabi.Tests.UnitTests.EventSourcing
+{
+	public class InMemoryEventRepositoryTests : IDisposable
+	{
+		private const string TestRoundAggregate = "TestRoundAggregate";
+		private readonly TimeSpan _semaphoreWaitTimeout = TimeSpan.FromSeconds(20);
+
+		public InMemoryEventRepositoryTests(ITestOutputHelper output)
+		{
+			TestEventRepository = new TestInMemoryEventRepository(output);
+			EventRepository = TestEventRepository;
+		}
+
+		private IEventRepository EventRepository { get; init; }
+		private TestInMemoryEventRepository TestEventRepository { get; init; }
+
+		[Fact]
+		public async Task AppendEvents_Zero_Async()
+		{
+			// Arrange
+			var events = Array.Empty<WrappedEvent>();
+
+			// Act
+			await EventRepository.AppendEventsAsync(nameof(TestRoundAggregate), "1", events);
+
+			// Assert
+			Assert.True((await EventRepository.ListEventsAsync(nameof(TestRoundAggregate), "1"))
+				.SequenceEqual(events));
+			Assert.True((await EventRepository.ListAggregateIdsAsync(nameof(TestRoundAggregate)))
+				.SequenceEqual(Array.Empty<string>()));
+		}
+
+		[Fact]
+		public async Task AppendEvents_One_Async()
+		{
+			// Arrange
+			var events = new[]
+			{
+				new WrappedEvent(1),
+			};
+
+			// Act
+			await EventRepository.AppendEventsAsync(nameof(TestRoundAggregate), "1", events);
+
+			// Assert
+			Assert.True((await EventRepository.ListEventsAsync(nameof(TestRoundAggregate), "1"))
+				.SequenceEqual(events));
+			Assert.True((await EventRepository.ListAggregateIdsAsync(nameof(TestRoundAggregate)))
+				.SequenceEqual(new[] { "1" }));
+		}
```

Comment:
I would write:

```suggestion
		[Fact]
		public async Task AppendEvents_One_Async()
		{
			// Arrange
			WrappedEvent[] expectedEvents = new[]
			{
				new WrappedEvent(1),
			};

			// Act
			await EventRepository.AppendEventsAsync(nameof(TestRoundAggregate), "MY_ID", expectedEvents);

			// Assert
			IReadOnlyList<WrappedEvent> actualEvents = await EventRepository.ListEventsAsync(nameof(TestRoundAggregate), "MY_ID");
			Assert.Equal(expectedEvents, actualEvents);

			IReadOnlyList<string> aggregateIds = await EventRepository.ListAggregateIdsAsync(nameof(TestRoundAggregate));
			Assert.Equal(new string[] { "MY_ID" }, aggregateIds);
		}
```

so that `1` is not used to mean an aggregate type and also sequence ID.

Reply:
I'm not rewriting tests.